app/models/scoped_event_feed.rb

```diff
@@ -0,0 +1,38 @@
+class ScopedEventFeed
+  attr_reader :root, :event_subjects, :page, :per
+  def initialize(root_type, root_id, page = 0, per = 20)
+    @root = root_type.classify.constantize.find(root_id)
+    @event_subjects = EventHierarchy.new(root).children
+    @page = page
+    @per = per
+  end
+
+  def events
+    if @events
+      @events
+    else
+      event_ids = event_subjects.group_by { |x| x.class.to_s }.inject([]) do |ids, (type, vals)|
```

Comment:
After talking to @susannasiebert we're going to try and generate some parameterized SQL here and see if its any more clear.